AstSemantics.md

```diff
@@ -248,6 +295,32 @@ The details of index space for local variables and their types will be further c
 e.g. whether locals with type `i32` and `i64` must be contiguous and separate from
 others, etc.
 
+## Global variables
+
+A *global variable* stores a single value of a fixed [value type](#types) and may be
+declared either *mutable* or *immutable*. This provides WebAssembly with memory
+locations that are disjoint from any [linear memory](#linear-memory) and thus
+cannot be arbitrarily aliased as bits.
+
```

Comment:
Great, see also https://github.com/WebAssembly/design/issues/333  These might also address some code path selection based on features.

But if only immutable globals are need then perhaps just implements these and not mutable globals?

Do we know if the compiler really need un-aliased locations, or is the linear memory just fine? 

Problem with mutable globals is that they need a pointer outside of the linear memory, increasing register pressure. The 32-bit x86 might be an exception where the offset can be baked in, but with code sharing even that would not be possible.